Food_Tracker_App/app/src/main/java/com/example/foodtracker/ui/recipes/RecipeDisplay.java

```diff
@@ -1,22 +1,103 @@
 
+import androidx.annotation.NonNull;
 import androidx.appcompat.app.AppCompatActivity;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
 
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
 import android.os.Bundle;
+import android.view.View;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.ListView;
+import android.widget.TextView;
+
 import com.example.foodtracker.R;
+import com.example.foodtracker.model.ingredient.Ingredient;
 import com.example.foodtracker.model.MenuItem;
+import com.example.foodtracker.model.recipe.Recipe;
 import com.example.foodtracker.ui.NavBar;
+import com.example.foodtracker.ui.TopBar;
 
 public class RecipeDisplay extends AppCompatActivity {
 
+    private TextView recipeTitle;
+    private TextView recipePrepTime;
+    private TextView recipeServings;
+    private TextView recipeCategory;
+    private TextView recipeComment;
+    //private ListView recipeIngredients;
+    private RecyclerView recipeIngredients;
+    private Button deleteRecipeButton;
+    private Button editRecipeButton;
+    private ImageView recipeImage;
+    private Uri imageURI;
+
+    //private ArrayAdapter<Ingredient> adapter;
+    private RecipeIngredientsRecyclerViewAdapter adapter;
+    private Recipe recipe;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.recipe_display);
 
+        Intent intent = getIntent();
+        recipe = (Recipe) intent.getSerializableExtra("RECIPE");
+
+        //adapter = new CustomList(this, recipe.getIngredients());
+        adapter = new RecipeIngredientsRecyclerViewAdapter(this, recipe.getIngredients());
+
+        recipeTitle = findViewById(R.id.recipeDisplayTitle);
+        recipePrepTime = findViewById(R.id.recipeDisplayPrepTime);
+        recipeServings = findViewById(R.id.recipeDisplayServings);
+        recipeCategory = findViewById(R.id.recipeDisplayCategory);
+        recipeComment = findViewById(R.id.recipeDisplayComment);
+        recipeIngredients = findViewById(R.id.recipeDisplayIngredientList);
+        deleteRecipeButton = findViewById(R.id.recipeDeleteButton);
+        editRecipeButton = findViewById(R.id.recipeEditButton);
+        recipeImage = findViewById(R.id.recipe_display_image);
+
+        recipeTitle.setText(recipe.getTitle());
+        recipePrepTime.setText("Preperation Time: " + String.valueOf(recipe.getPrepTime()));
+        recipeServings.setText("Servings: " + String.valueOf(recipe.getServings()));
+        recipeCategory.setText("Category: " + recipe.getCategory());
+        recipeComment.setText("Comment:\n\n" + recipe.getComment());
+//        if (!recipe.getImage().isEmpty()) {
+//            imageURI = Uri.parse(recipe.getImage());
+//            recipeImage.setImageURI(imageURI);
+//        }
```

Comment:
We may need to find a different way to store the image if this doesn't work, maybe create an issue for it if you are not going to implement it here

Reply:
I added an issue to the backlog. This was the error message that I got:
Caused by: java.lang.SecurityException: Permission Denial: opening provider com.android.providers.media.MediaDocumentsProvider from ... requires that you obtain access using ACTION_OPEN_DOCUMENT or related APIs.

Reply:
Yeah, I ended up taking a look at this because I was curious. It looks like the API to get the image and URI gives a permission that is lost on restart... which means that storing the URI is not super helpful in the end. We can probably store something else that represents the image like [here](https://stackoverflow.com/a/13563230). I will add this info to the issue and try it out on my end maybe